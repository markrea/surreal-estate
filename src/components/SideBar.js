import React from "react";
import { Link } from "react-router-dom"
import "../styles/SideBar.css";

const SideBar = () => {
  return (
    <div>
      <Link className="sidebar-link" to={`/?query={"city":"Manchester"}`}>Manchester</Link>
      <Link className="sidebar-link" to={`/?query={"city":"Liverpool"}`}>Liverpool</Link>
      <Link className="sidebar-link" to={`/?query={"city":"Sheffield"}`}>Sheffield</Link>
      <Link className="sidebar-link" to={`/?query={"city":"Leeds"}`}>Leeds</Link>
    </div>
  );
};


export default SideBar;
