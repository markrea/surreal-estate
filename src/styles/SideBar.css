input {
    width: 100px;
    height: 20px;
    background-color: white;
}
.search-bar {
    background-color: black;
}