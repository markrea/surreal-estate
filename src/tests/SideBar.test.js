import React from "react";
import { render } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import SideBar from "../components/SideBar";

describe("SideBar", () => {
  test("the component renders correctly against snapshot", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Router history={history}>
        <SideBar />
      </Router>
    );
    const component = asFragment();

    expect(component).toMatchSnapshot();
  });
});