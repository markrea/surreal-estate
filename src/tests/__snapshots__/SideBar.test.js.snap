// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar the component renders correctly against snapshot 1`] = `
<DocumentFragment>
  <div>
    <a
      class="sidebar-link"
      href="/?query={\\"city\\":\\"Manchester\\"}"
    >
      Manchester
    </a>
    <a
      class="sidebar-link"
      href="/?query={\\"city\\":\\"Liverpool\\"}"
    >
      Liverpool
    </a>
    <a
      class="sidebar-link"
      href="/?query={\\"city\\":\\"Sheffield\\"}"
    >
      Sheffield
    </a>
    <a
      class="sidebar-link"
      href="/?query={\\"city\\":\\"Leeds\\"}"
    >
      Leeds
    </a>
  </div>
</DocumentFragment>
`;
